packages/go/analysis/ad/esc9.go

```diff
@@ -106,15 +106,76 @@ func PostADCSESC9a(ctx context.Context, tx graph.Transaction, outC chan<- analys
 		}
 
 		results.Each(func(value uint32) bool {
-			if !channels.Submit(ctx, outC, analysis.CreatePostRelationshipJob{
+			return channels.Submit(ctx, outC, analysis.CreatePostRelationshipJob{
 				FromID: graph.ID(value),
 				ToID:   domain.ID,
 				Kind:   ad.ADCSESC9a,
-			}) {
-				return false
+			})
+		})
+
+		return nil
+	}
+}
+
+func PostADCSESC9b(ctx context.Context, tx graph.Transaction, outC chan<- analysis.CreatePostRelationshipJob, groupExpansions impact.PathAggregator, eca, domain *graph.Node, cache ADCSCache) error {
+	results := cardinality.NewBitmap32()
+
+	if canAbuseWeakCertBindingRels, err := FetchCanAbuseWeakCertBindingRels(tx, eca); err != nil {
+		if graph.IsErrNotFound(err) {
+			return nil
+		}
+
+		return err
+	} else if len(canAbuseWeakCertBindingRels) == 0 {
+		return nil
+	} else if publishedCertTemplates, ok := cache.PublishedTemplateCache[eca.ID]; !ok {
+		return nil
+	} else {
+		for _, template := range publishedCertTemplates {
+			if valid, err := isCertTemplateValidForESC9b(template); err != nil {
+				if !errors.Is(err, graph.ErrPropertyNotFound) {
+					log.Errorf("Error checking cert template validity for template %d: %v", template.ID, err)
+				} else {
+					log.Debugf("Error checking cert template validity for template %d: %v", template.ID, err)
+				}
+			} else if !valid {
+				continue
+			} else if certTemplateControllers, ok := cache.CertTemplateControllers[template.ID]; !ok {
+				log.Debugf("Failed to retrieve controllers for cert template %d from cache", template.ID)
+				continue
+			} else if ecaControllers, ok := cache.EnterpriseCAEnrollers[eca.ID]; !ok {
```

Comment:
Should move this call out of the loop, since it'll be the same for every iteration, small performance optimization